pkg/solana/fees/block_history.go

```diff
@@ -99,39 +121,161 @@ func (bhe *blockHistoryEstimator) readRawPrice() uint64 {
 }
 
 func (bhe *blockHistoryEstimator) calculatePrice(ctx context.Context) error {
+	switch bhe.estimationMethod {
+	case LatestBlockEstimator:
+		return bhe.calculatePriceFromLatestBlock(ctx)
+	case MultipleBlocksEstimator:
+		return bhe.calculatePriceFromMultipleBlocks(ctx, bhe.cfg.BlockHistoryDepth())
+	default:
+		return fmt.Errorf("unknown estimation method")
+	}
+}
+
+func (bhe *blockHistoryEstimator) calculatePriceFromLatestBlock(ctx context.Context) error {
 	// fetch client
 	c, err := bhe.client.Get()
 	if err != nil {
-		return fmt.Errorf("failed to get client in blockHistoryEstimator.getFee: %w", err)
+		return fmt.Errorf("failed to get client: %w", err)
 	}
 
 	// get latest block based on configured confirmation
 	block, err := c.GetLatestBlock(ctx)
 	if err != nil {
-		return fmt.Errorf("failed to get block in blockHistoryEstimator.getFee: %w", err)
+		return fmt.Errorf("failed to get block: %w", err)
 	}
 
 	// parse block for fee data
 	feeData, err := ParseBlock(block)
 	if err != nil {
-		return fmt.Errorf("failed to parse block in blockHistoryEstimator.getFee: %w", err)
+		return fmt.Errorf("failed to parse block: %w", err)
 	}
 
 	// take median of returned fee values
 	v, err := mathutil.Median(feeData.Prices...)
 	if err != nil {
-		return fmt.Errorf("failed to find median in blockHistoryEstimator.getFee: %w", err)
+		return fmt.Errorf("failed to find median: %w", err)
 	}
 
 	// set data
 	bhe.lock.Lock()
 	bhe.price = uint64(v) // ComputeUnitPrice is uint64 underneath
 	bhe.lock.Unlock()
 	bhe.lgr.Debugw("BlockHistoryEstimator: updated",
+		"method", bhe.estimationMethod.String(),
 		"computeUnitPrice", v,
 		"blockhash", block.Blockhash,
 		"slot", block.ParentSlot+1,
 		"count", len(feeData.Prices),
 	)
+
+	return nil
+}
+
+func (bhe *blockHistoryEstimator) calculatePriceFromMultipleBlocks(ctx context.Context, desiredBlockCount uint64) error {
+	if desiredBlockCount == 0 {
+		return fmt.Errorf("desiredBlockCount is zero")
+	}
+
+	if desiredBlockCount > MAX_BLOCK_HISTORY_DEPTH {
+		// Limit the desired block count to maxblockHistoryDepth
+		desiredBlockCount = MAX_BLOCK_HISTORY_DEPTH
```

Comment:
Maybe we should log a warning here

Reply:
The thing is this code executes once every `BlockHistoryPollPeriod`, so it may become really noisy. 

Wdyt about refactoring and checking/logging during initialization?

Reply:
I think it might be ok not setting a max block history unless during testing we find some sort of definitive upper limit. Otherwise, we could make a recommendation to CCIP if they want to set a high value to not go over a certain value but we shouldn't make it a hard requirement in the code.

Reply:
> Wdyt about refactoring and checking/logging during initialization?

If we do want to keep an upper bound, I think it's better to move this to initialization. But instead of just logging a warning, we should fail initialization so the person running the node can update the configs accordingly.

Reply:
I'll remove it for now. Let's see what happens when testing with different values

I got a limited around 20slots/5s for the public rpc endpoint for devnet

I'm not sure what the limit will be in terms of mainnet